storm-core/src/jvm/backtype/storm/utils/Utils.java

```diff
@@ -764,5 +778,30 @@ public static long zipFileSize(File myFile) throws IOException{
     public static double zeroIfNaNOrInf(double x) {
         return (Double.isNaN(x) || Double.isInfinite(x)) ? 0.0 : x;
     }
+
+    /**
+     * parses the arguments to extract jvm heap memory size.
+     * @param input
+     * @param defaultValue
+     * @return the value of the JVM heap memory setting in a java command.
+     */
+    public static Double parseWorkerChildOpts(String input, Double defaultValue) {
+        if (input != null) {
+            Pattern optsPattern = Pattern.compile("Xmx[0-9]+m");
```

Comment:
This is not good enough.  It is not just an 'm' that is supported.

From http://docs.oracle.com/javase/7/docs/technotes/tools/windows/java.html

```
-Xmxn

    Specifies the maximum size, in bytes, of the memory allocation pool. This value must a multiple of 1024 greater than 2 MB. Append the letter k or K to indicate kilobytes, or m or M to indicate megabytes. The default value is chosen at runtime based on system configuration.
```

I have also seen 'g' and 'G' work for gigabytes too.